Provider/src/FirebirdSql.Data.UnitTests/FbConnectionTests.cs

```diff
@@ -303,11 +303,32 @@ public void DatabaseTriggersTest()
 			}
 		}
 
-		#endregion
-
-		#region Methods
-
-		public FbTransaction BeginTransaction(IsolationLevel level)
+        [Test]
+        public void LogonUsernameTest()
+        {
+            string loggedUser = string.Empty;
+            using (var conn = new FbConnection(BuildConnectionString(FbServerType)))
+            {
+                conn.Open();
+                using (var transaction = conn.BeginTransaction())
+                {
+                    using (var command = conn.CreateCommand())
+                    {
+                        command.Transaction = transaction;
+                        command.CommandText = "select CURRENT_USER from RDB$DATABASE";
+                        loggedUser = (string)command.ExecuteScalar();
```

Comment:
i did it externally to don't rollback the transaction (less garbage it's better :) ), i will put the assert inside if you prefer